pkg/pillar/cmd/downloader/downloader.go

```diff
@@ -404,8 +405,11 @@ func handleModify(ctx *downloaderContext, key string,
 		status.IsContainer = config.IsContainer
 		publishDownloaderStatus(ctx, status)
 	}
-	// If RefCount from zero to non-zero then do install
-	if status.RefCount == 0 && config.RefCount != 0 {
+	if status.PendingAdd {
+		log.Infof("handleModify: Download is already in progress for %s", config.Name)
+		return
+	} else if status.RefCount == 0 && config.RefCount != 0 {
```

Comment:
Stylistically confusing to have a return then an else. This line can be just an if statement.

Reply:
Resolved.